src/logic/calculate.js

```diff
@@ -0,0 +1,42 @@
+import operate from './operate';
+
+const calculate = (calculatorData, buttonName) => {
+  const { total, next, operation } = calculatorData;
+
+  switch (buttonName) {
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+      if (!total) {
+        return { total: buttonName, next, operation };
+      }
+      return { total: total + buttonName, next, operation };
+
+    case '+/-':
+      return { total: -total, next: -next, operation };
```

Comment:
we're not to review the logic/body of this function, because it will catchup with youo n milestone 5. This is just a small warning ;)

It's nicely setup, destructured, imported